lib/rubygems/rdoc.rb

```diff
@@ -6,8 +6,17 @@
   require "rdoc/rubygems_hook"
   module Gem
     RDoc = ::RDoc::RubygemsHook
+
+    ##
+    # Returns whether RDoc defines its own install hooks through a RubyGems
+    # plugin. This and whatever is guarded by it can be removed once no
+    # supported Ruby ships with RDoc older than 6.9.0.
+
+    def self.rdoc_hooks_defined_via_plugin?
+      Gem::Version.new(::RDoc::VERSION) >= Gem::Version.new("6.9.0")
```

Comment:
Or we can check `RDoc::RubyGemsHook` (not `RubygemsHook`):

```suggestion
      ::RDoc.const_defined?(:RubyGemsHook)
```

(Either way is fine.)

Reply:
Yeah, I thought of that but I ended using an explicit check over feature detection. I would add a note about the version introducing the constant anyways so that we don't forget to remove the code when the time comes, so I figured it was easier to check the version directly.

Reply:
It makes sense.